test/types/range/lydia/badRange-string-lower.good

```diff
@@ -0,0 +1,4 @@
+badRange-string-lower.chpl:1: In function 'main':
+badRange-string-lower.chpl:2: error: lower bound of param for loop must be an int, uint, or bool
+badRange-string-lower.chpl:2: error: lower bound and upper bound of param for loop expected to be of the same type
+badRange-string-lower.chpl:2: error: lower bound and stride of param for loop expected to be of the same type
```

Comment:
What would it take to make this less about the param for loop and more about the fact that the range is completely bogus? (i.e., you can't have a range from string to integer).

Reply:
I could change the error message to refer to the range instead, or it would mean solving the problem additionally in a separate spot.  At the time the internal error was occurring, we were in a ParamForLoop AST node and it looks like it doesn't separate the range out into a separate construct until after the fact (if I'm reading the code correctly).  But we technically know it's a range.

Reply:
Can we store param ranges right now?

Reply:
> Can we store param ranges right now?

You're right that we can't and that the current compiler architecture represents this more as a special loop form with explicit low and high bounds rather than a general loop over a param iterator.  But I don't think that's the perspective we want to present to the user.

Put another way: `""..2` would be a problematic range expression in any context, not just param for loops.  I think that flagging the error where you are makes sense given the current code organization, but I think we should complain about the range in a way that's similar to what we do (or would like to do) in other contexts rather than taking about param for loop bounds.  For example, this program:

```chapel
var r = ""..10;
```

generates the (so-so) error message:

```
testit.chpl:1: error: Bounds of 'low..high' must be integers of compatible types.
```

Could we simply print the same thing for these cases?  (or, if we wanted to improve it, maybe "Range bounds must be of compatible types."?) 

(I think the current message is so-so in that it doesn't use the "range" type by name and that ranges now also support enums and bools and may support other types in the future).

Reply:
>  But I don't think that's the perspective we want to present to the user.

That seems reasonable to me, yeah.

> Could we simply print the same thing for these cases? (or, if we wanted to improve it, maybe "Range bounds must be of compatible types."?)

So I should adjust the old error message to match this new one, and use it for my new messages, yes?  Not a problem, just making sure I'm understanding